phoenix-core/src/it/java/org/apache/phoenix/end2end/CDCDefinitionIT.java

```diff
@@ -0,0 +1,328 @@
+        assertCDCState(conn, cdcName, null, 3);
+        assertNoResults(conn, cdcName);
+
+        try {
+            conn.createStatement().execute(cdc_sql);
+            fail("Expected to fail due to duplicate index");
+        } catch (SQLException e) {
+            assertEquals(SQLExceptionCode.TABLE_ALREADY_EXIST.getErrorCode(), e.getErrorCode());
+            assertTrue(e.getMessage().endsWith(cdcName));
+        }
+
+        conn.createStatement().execute("CREATE CDC IF NOT EXISTS " + cdcName + " ON " + tableName +
+                " INCLUDE (pre, post) INDEX_TYPE=g");
+
+        cdcName = generateUniqueName();
+        cdc_sql = "CREATE CDC " + cdcName + " ON " + tableName + " INCLUDE (pre, post)";
+        createCDCAndWait(conn, tableName, cdcName, cdc_sql, PTable.IndexType.UNCOVERED_GLOBAL);
+        assertCDCState(conn, cdcName, "PRE,POST", 3);
+        assertPTable(cdcName, new HashSet<>(
+                Arrays.asList(PTable.CDCChangeScope.PRE, PTable.CDCChangeScope.POST)), tableName,
+                datatableName);
+        assertNoResults(conn, cdcName);
+
+        cdcName = generateUniqueName();
+        cdc_sql = "CREATE CDC " + cdcName + " ON " + tableName + " INDEX_TYPE=l";
+        createCDCAndWait(conn, tableName, cdcName, cdc_sql, PTable.IndexType.LOCAL);
+        assertCDCState(conn, cdcName, null, 2);
+        assertPTable(cdcName, null, tableName, datatableName);
+        assertNoResults(conn, cdcName);
+
+        conn.close();
+    }
+
+    @Test
+    public void testCreateWithSalt() throws Exception {
+        // Indexes on views don't support salt buckets and is currently silently ignored.
+        if (forView) {
+            return;
+        }
+
+        // {data table bucket count, CDC bucket count}
+        Integer[][] saltingConfigs = new Integer[][] {
+                new Integer[]{null, 2},
+                new Integer[]{0, 2},
+                new Integer[]{4, null},
+                new Integer[]{4, 1},
+                new Integer[]{4, 0},
+                new Integer[]{4, 2}
+        };
+
+        for (Integer[] saltingConfig: saltingConfigs) {
+            try (Connection conn = newConnection()) {
+                String tableName = generateUniqueName();
+                createTable(conn, "CREATE TABLE  " + tableName +
+                                " ( k INTEGER PRIMARY KEY, v1 INTEGER, v2 DATE)",
+                                null, false, saltingConfig[0], false, null);
+                assertSaltBuckets(conn, tableName, saltingConfig[0]);
+
+                String cdcName = generateUniqueName();
+                String cdc_sql = "CREATE CDC " + cdcName + " ON " + tableName;
+                createCDCAndWait(conn, tableName, cdcName, cdc_sql, null,
+                        saltingConfig[1], null);
+                try {
+                    assertCDCState(conn, cdcName, null, 3);
+                    // Index inherits table salt buckets.
```

Comment:
This comment is not right.

Reply:
It is meant for the next line, will move it.